src/tracer/wrap_syscall.rs

```diff
@@ -322,7 +322,17 @@ impl KvmRunWrapper {
             WaitStatus::Stopped(pid, signal) => {
                 return self.stopped(pid, &signal);
             }
-            WaitStatus::Exited(_, status) => bail!("process exited with: {}", status),
+            WaitStatus::Exited(pid, status) => {
+                if let Some(index) = self
+                    .threads
+                    .iter()
+                    .position(|thread| thread.ptthread.tid == pid)
+                {
+                    self.threads.remove(index);
```

Comment:
Do you mean? 
```suggestion
                    // remove and shift others to left
                    self.threads.remove(idx);
                    // shift idx if it was shifted
                    if idx > self.process_idx {
                        self.process_idx -= 1;
                    }

```

Reply:
yes (`>` -> `<`)
```suggestion
                    // remove and shift others to left
                    self.threads.remove(idx);
                    // shift idx if it was shifted
                    if idx < self.process_idx {
                        self.process_idx -= 1;
                    }
```

Reply:
it fixes the crash but it looks like it leaves the device unresponsive. Not sure if this issue is related.